pkg/k8s/client.go

```diff
@@ -121,51 +121,40 @@ func createClient(config *rest.Config, cs kubernetes.Interface) error {
 	return err
 }
 
-func createDefaultClient() (rest.Interface, func(), error) {
-	restConfig, err := CreateConfig()
-	if err != nil {
-		return nil, nil, fmt.Errorf("unable to create k8s client rest configuration: %s", err)
-	}
+func createDefaultClient(c *rest.Config, httpClient *http.Client) (rest.Interface, error) {
+	restConfig := *c
 	restConfig.ContentConfig.ContentType = `application/vnd.kubernetes.protobuf`
 
-	closeAllConns := setDialer(restConfig)
-
-	createdK8sClient, err := kubernetes.NewForConfig(restConfig)
+	createdK8sClient, err := kubernetes.NewForConfigAndClient(&restConfig, httpClient)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
-	err = createClient(restConfig, createdK8sClient)
+	err = createClient(&restConfig, createdK8sClient)
 	if err != nil {
-		return nil, nil, fmt.Errorf("unable to create k8s client: %s", err)
+		return nil, fmt.Errorf("unable to create k8s client: %s", err)
 	}
 
 	k8sCLI.Interface = createdK8sClient
 
-	createK8sWatcherCli, err := watcher_client.NewForConfig(restConfig)
+	createK8sWatcherCli, err := watcher_client.NewForConfigAndClient(&restConfig, httpClient)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
 	k8sWatcherCLI.Interface = createK8sWatcherCli
 
-	return createdK8sClient.RESTClient(), closeAllConns, nil
+	return createdK8sClient.RESTClient(), nil
 }
 
-func createDefaultCiliumClient() (func(), error) {
-	restConfig, err := CreateConfig()
+func createDefaultCiliumClient(c *rest.Config, httpClient *http.Client) error {
+	createdCiliumK8sClient, err := clientset.NewForConfigAndClient(c, httpClient)
 	if err != nil {
-		return nil, fmt.Errorf("unable to create k8s client rest configuration: %s", err)
-	}
-
-	closeAllConns := setDialer(restConfig)
-	createdCiliumK8sClient, err := clientset.NewForConfig(restConfig)
-	if err != nil {
-		return nil, fmt.Errorf("unable to create k8s client: %s", err)
+		return fmt.Errorf("unable to create k8s client: %s", err)
 	}
 
 	k8sCiliumCLI.Interface = createdCiliumK8sClient
 
-	return closeAllConns, nil
+	return nil
 }
 
 func createAPIExtensionsClient() error {
```

Comment:
any reason not reusing http client for extensions client ? or is there no significant benefit?

Reply:
oh this was something that I missed. Thanks for spotting it :male_detective: